### PR TITLE
feat(Tag): add new Functional Tag

### PR DESCRIPTION
**Pull request by akim316**
Closes carbon-design-system/carbon-components-react#865

Moved over `carbon-components-react` changes to `carbon-addons-cloud`.

Stateful Tag component that is removable and has a onRemove callback

### CODE REVIEW

File: src/components/Tag/_colors.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+$color__navy-gray-10: #d0dcef !default;
```
[the remark this answers is not in the document]
Comment (akim316) [reply src/components/Tag/_colors.scss]: These are custom colors Virginia has outlined, should I name them something else or move it?

[the remark this answers is not in the document]
Comment (joshblack) [reply src/components/Tag/_colors.scss]: Typically, we try and map colors to color role tokens. Can see a description of them here: http://carbondesignsystem.com/style/colors/usage#color-roles let me know if that's not the case here!

**Comment by akim316** [comment]
https://functional-tag-storybook.mybluemix.net/?selectedKind=Tag&selectedStory=Default%20Tag&full=0&down=1&left=1&panelRight=0&downPanel=jbovenschen%2Fstorybook-addon-a11y%2Faddon-panel
Temp link for Storybook with Tag ^

**Comment by vsnichols** [comment]
I approve the changes - thanks @joshblack @akim316

**Comment by joshblack** [comment]
Awesome, thanks @vsnichols! Just waiting for some of the comments to be addressed. @akim316 Do you also want to get someone from your team to approve the PR as well? Typically we'll have two reviewers sign off before merging 👍 Can also get another Carbon dev to review, if needed!

File: src/components/Tag/_colors.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (vsnichols) [reply src/components/Tag/_colors.scss]: Hi Josh - this tag uses custom colors intentionally to be very close to the background page color in IBM Cloud but slightly darker. I believe Tayler and Anna are aware that it uses custom colors.

[the remark this answers is not in the document]
Comment (joshblack) [reply src/components/Tag/_colors.scss]: Cool, sounds good! cc @aagonzales should we do anything special for color roles here, or just have them be custom colors?

**Pull request merged**